src/plugins/mer/mersettings.cpp

```diff
@@ -115,6 +115,74 @@ void MerSettings::setRpmValidationByDefault(bool byDefault)
     emit s_instance->rpmValidationByDefaultChanged(s_instance->m_rpmValidationByDefault);
 }
 
+QHash<QString, MerEmulatorDeviceModel> MerSettings::customEmulatorDeviceModels()
+{
+    Q_ASSERT(s_instance);
+
+    return s_instance->m_customEmulatorDeviceModels;
+}
+
+bool MerSettings::isCustomEmulatorDeviceModel(const QString &name)
+{
+    Q_ASSERT(s_instance);
+
+    return s_instance->m_customEmulatorDeviceModels.contains(name);
+}
+
+MerEmulatorDeviceModel MerSettings::customEmulatorDeviceModel(const QString &name)
+{
+    Q_ASSERT(s_instance);
+
+    return s_instance->m_customEmulatorDeviceModels.value(name);
+}
+
+void MerSettings::setCustomEmulatorDeviceModel(const QString &name, const MerEmulatorDeviceModel &v)
+{
+    Q_ASSERT(s_instance);
+
+    s_instance->m_customEmulatorDeviceModels[name] = v;
+}
+
+bool MerSettings::createCustomEmulatorDeviceModel(const QString &name)
+{
+    Q_ASSERT(s_instance);
+
+    const auto models = MerSettings::customEmulatorDeviceModels();
+    if (models.contains(name))
+        return false;
+
+    QVariantMap map;
+    map.insert(QLatin1String(Constants::MER_DEVICE_MODEL_NAME), name);
+    map.insert(QLatin1String(Constants::MER_DEVICE_MODEL_DISPLAY_RESOLUTION), QRect(QPoint(0, 0), QSize(500, 900)));
+    map.insert(QLatin1String(Constants::MER_DEVICE_MODEL_DISPLAY_SIZE), QRect(QPoint(0, 0), QSize(50, 90)));
+    map.insert(QLatin1String(Constants::MER_DEVICE_MODEL_DCONF_DB), QString());
+    MerEmulatorDeviceModel emulator;
+    emulator.fromMap(map);
```

Comment:
This looks like a duplication of the serialization code. Wouldn't it be more logical to init the structure directly? The name "emulator" is misleading.

Reply:
Removed.